objects/auth_factor.json

```diff
@@ -0,0 +1,96 @@
+{
+  "caption": "Authentication Factor",
+  "description": "Information on factors used to authenticate.",
+  "name": "authentication factor",
+  "extends": "object",
+  "attributes": {
+    "factor_device": {
+      "description": "Device used to complete the authentication request.",
+      "requirement": "recommended"
+    },
+    "factor_type": {
```

Comment:
So for factor type, I decided to follow more of a "software" vs "hardware" token division instead of "totp" vs "hotp". I learned that a lot of factors can fall under "totp" or "hotp" that we'd want split out, e.g. SMS is technically hotp and webauthn can be either hotp or totp. 

I left "hardware tokens" as a group, though debating pulling out USB based tokens and leaving "hardware tokens" as a catch all for the other types. 

Software tokens naturally split out into separate factors. This includes OTP apps, push notifications, SMS, email, etc. Software tokens is too vague on its own. 

I did include two booleans, `is_totp` and `is_hotp` to capture this information about the different factors.